torch/_jit_internal.py

```diff
@@ -796,6 +796,22 @@ def is_dict(ann):
         (getattr(ann, '__origin__', None) is Dict or
             getattr(ann, '__origin__', None) is dict)
 
+def is_union(ann):
```

Comment:
The commented lines can be deleted I think.

Also there is a chance to clean up redundant code inside `is_optional` to reuse `is_union`